lib/vector-common/src/internal_event/cached_event.rs

```diff
@@ -22,7 +23,7 @@ pub struct RegisteredEventCache<T, Event: RegisterTaggedInternalEvent> {
     fixed_tags: T,
     cache: Arc<
         RwLock<
-            BTreeMap<
+            HashMap<
```

Comment:
Is there a motivation to prefer one over the other? I don't disagree with this change, just wondering.

Reply:
Ah. I was experimenting with removing the `Ord` and then just didn't change this back when I realised I needed the `Ord` anyway. I can't think of any clear reason why one would be better than the other.